# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1374768158 -7200
#      Thu Jul 25 18:02:38 2013 +0200
# Node ID 7a67ab7c47714827ff6886327942e65f1655c2aa
# Parent  18137205f57db176574be1f2f6954f68ab3d89ee
Initialize uimenu objects for the fltk toolkit.

* libinterp/dldfcn/__init_fltk__.cc (fltk_graphics_toolkit): Modify the
initialze method to include uimenu objects.

diff --git a/libinterp/dldfcn/__init_fltk__.cc b/libinterp/dldfcn/__init_fltk__.cc
--- a/libinterp/dldfcn/__init_fltk__.cc
+++ b/libinterp/dldfcn/__init_fltk__.cc
@@ -1840,17 +1840,28 @@ public:
       input_event_hook_fcn_id ()
   { }
 
   ~fltk_graphics_toolkit (void) { }
 
   bool is_valid (void) const { return true; }
 
   bool initialize (const graphics_object& go)
-    { return go.isa ("figure"); }
+  {
+    if (go.isa ("figure")
+        || go.isa ("uimenu"))
+      {
+        if (go.isa ("uimenu"))
+          update (go, uimenu::properties::ID_LABEL);
+
+        return true;
+      }
+
+    return false;
+  }
 
   void finalize (const graphics_object& go)
   {
     if (go.isa ("figure"))
       {
         octave_value ov = go.get (caseless_str ("__plot_stream__"));
 
         if (! ov.is_empty ())
